src/cmd/services/m3comparator/main/querier.go

```diff
@@ -147,15 +155,35 @@ func (q *querier) FetchCompressed(
 	}, cleanup, nil
 }
 
-func (q *querier) generateRandomIters(
+func (q *querier) generateRandomSeries(
+	query *storage.FetchQuery,
+	blockSize time.Duration,
+) ([]series, error) {
+	var (
+		start = query.Start.Truncate(blockSize)
+		end   = query.End.Truncate(blockSize).Add(blockSize)
+	)
+
+	metricNameTag := q.opts.tagOptions.MetricName()
+	for _, matcher := range query.TagMatchers {
+		if bytes.Equal(metricNameTag, matcher.Name) {
+			if matched, _ := regexp.Match(`^multi_\d+$`, matcher.Value); matched {
+				return q.generateMultiSeriesMetrics(string(matcher.Value), start, end, time.Second*30, blockSize)
```

Comment:
Maybe make `resolution` also a field on `querier` since that may want to be changed later too?